### PR TITLE
Adds ENV vars for configuring the BB server

### LINKED ISSUE #287: Server settings should be configured by environment variables
There are some settings that are handled on a YML file: 

https://github.com/decidim/decidim-bulletin-board/blob/develop/bulletin_board/server/config/settings.yml

We should move this to the secrets.yml file and pick it up with environment variables to be consistent with `decidim/decidim`, ease the configuration and also to play nicer with Docker deployments.

### PR DESCRIPTION
**Pull request by microstudi**
fixes #287 
supersedes #301

### CODE REVIEW

File: bulletin_board/server/.rubocop.yml
Comment (andreslucena) [inline bulletin_board/server/.rubocop.yml:1]: Can you leave it out with `develop` and bump the rubocop version 🙏🏽 ?

File: bulletin_board/server/spec/commands/create_election_spec.rb [hunk 1]
```diff
@@ -87,7 +87,27 @@
     it_behaves_like "create election fails"
 
     it "broadcasts invalid" do
-      expect { subject }.to broadcast(:invalid, "Election should start at least in 2 hours from now.")
+      expect { subject }.to broadcast(:invalid, "Election should start at least in 1 hours from now.")
+    end
+
+    context "when HOURS_BEFORE_CREATE_ELECTION is set" do
```
Comment (andreslucena) [inline bulletin_board/server/spec/commands/create_election_spec.rb]: As we're not doing the test with the ENV VAR, I'd prefer to not reference to that in the example title

```suggestion
    context "when the `hours before` the election creation is set" do
```

**Review by andreslucena — CHANGES_REQUESTED** [review]
Two small changes, let me know if you prefer that I handle the rubocop update

File: bulletin_board/server/.rubocop.yml
Comment (microstudi) [reply bulletin_board/server/.rubocop.yml:1]: ok done. this forces to update many other files though

**Comment by andreslucena** [comment]
@microstudi there's a failing spec, can you check it out please?

**Comment by microstudi** [comment]
All done @andreslucena !

File: bulletin_board/server/.rubocop.yml [hunk 2]
```diff
@@ -20,3 +25,8 @@ Style/OpenStructUse:
 
 Rails/HasManyOrHasOneDependent:
   Enabled: false
+
+FactoryBot/FactoryAssociationWithStrategy:
+  Enabled: false
+
+
```
Comment (andreslucena) [inline bulletin_board/server/.rubocop.yml:32]:
```suggestion
```

**Review by andreslucena — APPROVED** [review]
As we've talked offline, you'll fix the empty line in the next PR, so I'll merge this one

**Pull request merged**